pygef/utils.py

```diff
@@ -124,7 +124,7 @@ def parse_zid_as_float(s):
     :param s: (str) String from which the zid is parsed.
     :return: ZID number.
     """
-    return parse_regex_cast(r"#ZID[=\s+]+[^,]*[,\s+]+([^,]+)", s, float, 1)
+    return parse_regex_cast(r"#ZID[=\s+]+[^,]*[,\s+]+([^?!,$|\s$]+)", s, float, 1)
 
```

Comment:
What was wrong?

Reply:
The z_id of some cpts was not parsed correctly, now it works better.

Reply:
For example the cpts of the project of Thomas, you can try the regex.